guides/plugins/plugins/elasticsearch/add-product-entity-extension-to-elasticsearch.md

```diff
@@ -0,0 +1,341 @@
+use Shopware\Core\Framework\DataAbstractionLayer\Field\ObjectField;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\OneToManyAssociationField;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\OneToOneAssociationField;
+use Shopware\Core\Framework\DataAbstractionLayer\FieldCollection;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\Flag\Runtime;
+
+class CustomExtension extends EntityExtension
+{
+    public function extendFields(FieldCollection $collection): void
+    {
+        //Add ApiAware flag to make this field searchable
+        $collection->add(
+            (new OneToOneAssociationField('oneToOneExampleExtension', 'id', 'product_id', OneToOneExampleExtensionDefinition::class, true))->addFlags(new ApiAware())
+        );
+        //Add ApiAware flag to make this field searchable
+        $collection->add(
+            (new OneToManyAssociationField('oneToManyExampleExtension', OneToManyExampleExtensionDefinition::class, 'product_id'))->addFlags(new ApiAware())
+        );
+        //Runtime fields are not searchable
+        $collection->add(
+            (new ObjectField('custom_string', 'customString'))->addFlags(new Runtime())
+        );
+    }
+
+    public function getDefinitionClass(): string
+    {
+        return ProductDefinition::class;
+    }
+}
+```
+{% endcode %}
+
+The entity definition `OneToManyExampleExtensionDefinition.php`.
+{% code title="<plugin root>/src/Extension/Content/Product/OneToManyExampleExtensionDefinition.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Extension\Content\Product;
+
+use Shopware\Core\Content\Product\ProductDefinition;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityDefinition;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\FkField;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\Flag\ApiAware;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\Flag\PrimaryKey;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\Flag\Required;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\IdField;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\ManyToOneAssociationField;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\StringField;
+use Shopware\Core\Framework\DataAbstractionLayer\FieldCollection;
+use Shopware\Core\Framework\DataAbstractionLayer\Entity;
+
+class OneToManyExampleExtensionDefinition extends EntityDefinition
+{
+    public const ENTITY_NAME = 'one_to_many_swag_example_extension';
+
+    public function getEntityName(): string
+    {
+        return self::ENTITY_NAME;
+    }
+
+    public function getEntityClass(): string
+    {
+        return Entity::class;
+    }
+
+    protected function defineFields(): FieldCollection
+    {
+        return new FieldCollection([
+            (new IdField('id', 'id'))->addFlags(new ApiAware(), new Required(), new PrimaryKey()),
+            new FkField('product_id', 'productId', ProductDefinition::class),
```

Comment:
I am pretty sure we will need also a ReferenceVersionField here for the product as its versionable

Reply:
Will work like this (the association will just not be versioned), but I will add it.